### PR TITLE
RELATED: RAIL-3383 Rework filterBar and layout to hooks customization API

### PR DESCRIPTION
**Pull request by no23reason**
JIRA: RAIL-3383

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4558/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6176/ : SUCCESS in 14m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/4559/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6177/ : SUCCESS in 14m 09s

File: libs/sdk-ui-dashboard/src/dashboard/types.ts [hunk 1]
```diff
@@ -229,6 +210,51 @@ export interface IDashboardProps {
...
+     *
+     * To access the necessary props in your custom component, use the {@link useDashboardAttributeFilterProps} hook.
+     * To fall back to the default implementation, use the {@link DefaultDashboardAttributeFilter} component.
+     */
+    DashboardAttributeFilterComponentFactory?: (
+        filter: IDashboardAttributeFilter,
+    ) => CustomDashboardAttributeFilterComponent | undefined;
+
+    /**
+     * Optionally specify component to use for rendering the attribute filters.
```
Comment (lupko) [inline libs/sdk-ui-dashboard/src/dashboard/types.ts:238]: date filters

Comment (no23reason) [reply libs/sdk-ui-dashboard/src/dashboard/types.ts]: Good catch, updated.

**Review by lupko — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1564

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1297/ : FAILURE in 2m 50s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1573/ : SUCCESS in 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/347/ : SUCCESS in 12m 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/6187/ : SUCCESS in 17m 14s

**Pull request merged**